src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/ECDsaCng.ImportExport.cs

```diff
@@ -0,0 +1,558 @@
+
+                    ECPoint qpoint = new ECPoint();
+                    ecParams.Q = qpoint;
+                    qpoint.X = Consume(ecBlob, ref offset, pBcryptBlob->cbFieldLength);
+                    qpoint.Y = Consume(ecBlob, ref offset, pBcryptBlob->cbFieldLength);
+
+                    if (includePrivateParameters)
+                    {
+                        ecParams.D = Consume(ecBlob, ref offset, pBcryptBlob->cbSubgroupOrder);
+                    }
+                }
+            }
+
+            return ecParams;
+        }
+
+        #endregion
+
+        /// <summary>
+        ///     Append "value" to the data already in blob.
+        /// </summary>
+        private static void Emit(byte[] blob, ref int offset, byte[] value)
+        {
+            Buffer.BlockCopy(value, 0, blob, offset, value.Length);
+            offset += value.Length;
+        }
+
+        /// <summary>
+        ///     Peel off the next "count" bytes in blob and return them in a byte array.
+        /// </summary>
+        private static byte[] Consume(byte[] blob, ref int offset, int count)
+        {
+            byte[] value = new byte[count];
+            Buffer.BlockCopy(blob, offset, value, 0, count);
+            offset += count;
+            return value;
+        }
+
+        /// <summary>
+        /// Get the ALG_ID from the given HashAlgorithmName
+        /// </summary>
+        private static Interop.BCrypt.ECC_CURVE_ALG_ID_ENUM GetHashAlgorithm(HashAlgorithmName name)
+        {
+            if (name == null || string.IsNullOrEmpty(name.Name))
+            {
+                return Interop.BCrypt.ECC_CURVE_ALG_ID_ENUM.BCRYPT_NO_CURVE_GENERATION_ALG_ID;
+            }
+
+            CRYPT_OID_INFO oid = OidInfo.FindOidInfo(
+                CryptOidInfoKeyType.CRYPT_OID_INFO_CNG_ALGID_KEY,
+                name.Name,
+                OidGroup.HashAlgorithm,
+                false);
+
+            if (oid.AlgId == -1)
+            {
+                throw new CryptographicException(SR.Cryptography_InvalidHashAlgorithmName); //todo sync with Cryptography_UnknownHashAlgorithm
+            }
+
+            return (Interop.BCrypt.ECC_CURVE_ALG_ID_ENUM)oid.AlgId;
+        }
+
+        private static bool ValidateKey(ECParameters parameters, out int cbKey)
+        {
+            cbKey = 0;
+
+            if (parameters.Curve == null || 
+                parameters.Q == null ||
+                parameters.Q.X == null ||
+                parameters.Q.Y == null)
+                return false;
+
+            cbKey = parameters.Q.X.Length;
+
+            if (cbKey == 0 ||
+                parameters.Q.Y.Length != cbKey ||
+                (parameters.D != null && parameters.D.Length != cbKey))
+                return false;
+
+            return true;
+        }
+
+        private static bool ValidateNamedCurve(ECNamedCurve curve)
+        {
+            if (curve.Oid == null ||
+                string.IsNullOrEmpty(curve.Oid.FriendlyName))
```

Comment:
Cleaning up some of the oid stuff now. Changed it to
            if (curve.Oid == null ||
                (string.IsNullOrEmpty(curve.Oid.FriendlyName) && string.IsNullOrEmpty(curve.Oid.FriendlyName)))